tutorials/hpc_clusters.md

```diff
@@ -0,0 +1,406 @@
+variable). Not many clusters do this. Some do it e.g. only for the interactive
+partitions (like the OP's one) so the first set of commands works, but the
+latter does not.
+
+## Approach 2: Add \--nv switch
+
+*This tests Gazebo in GLX mode connected to an existing X server, passing
+--nv to Singularity.*
+
+Now it's time to get a bit further and try running Singularity with the `--nv`
+switch which basically adds some GPU-related host libraries to the container.
+If you're on AMD GPUs, you should use `--rocm` instead, but that is not tested.
+Contributions are welcome!
+
+Test with almost the same command, just add `--nv` to singularity command:
+
+```bash
+user@cluster$ srun -p gpufast--gres=gpu:1 --pty bash -i
+user@gpu-node-1$ singularity exec --nv image.sif bash -c "echo $DISPLAY; gz sim -v4 -s -r sensors_demo.sdf & (gz topic -t /camera -e | cut -c -80) & sleep 30; kill %1; kill %2; sleep 2; kill -9 %1; kill -9 %2"
+```
+
+Also, try it on the non-interactive partition, adjusting `test.batch`
+appropriately.
+
+What can happen if the node's OS is much newer than the OS inside your image,
+is you'll get errors with GLIBC version mismatch. See
+https://github.com/apptainer/apptainer/issues/945#issuecomment-2096052536 for a
+workaround if that's your case.
+
+If both interactive and non-interactive commands work for you in this case, you
+are done (again, verify `RENDERER` in `ogre2.log`).
+
+**Explanation why this worked:** Most Singularity images do not install
+proprietary NVidia drivers inside. Instead, they rely on the host drivers, and
+the `--nv` switch bind-mounts all files related to the host drivers inside the
+container. So if there is a running X11 server, the programs from container
+can use it.
+
+## Approach 3: Try \--headless-rendering
+
+*This tests Gazebo in EGL mode using autodetected GPU, passing --nv to
+Singularity.*
+
+This is the mode described in tutorial \ref headless_rendering.
+
+If GLX does not work, let's try EGL backend:
+
+```bash
+user@cluster$ srun -p gpufast--gres=gpu:1 --pty bash -i
+user@gpu-node-1$ singularity exec --nv image.sif bash -c "gz sim -v4 -s -r --headless-rendering sensors_demo.sdf & (gz topic -t /camera -e | cut -c -80) & sleep 30; kill %1; kill %2; sleep 2; kill -9 %1; kill -9 %2"
+```
+
+And, again, try the same on a non-interactive partition.
+
+Check `RENDERER` in `ogre2.log` again.
+
+You can also have a more detailed look into `ogre2.log` and if this method
+works, you should see similar content:
+
+```bash
+01:13:59: GL3PlusRenderSystem::_createRenderWindow "OgreWindow(0)_0", 1x1 windowed  miscParams: FSAA=0 border=none contentScalingFactor=1.000000 gamma=Yes parentWindowHandle=0 stereoMode=Frame Sequential
+01:13:59: Trying to init device: EGL_EXT_device_drm EGL_EXT_device_drm_render_node #0 /dev/dri/card1...
+01:13:59: Created GL 4.5 context for device EGL_EXT_device_drm EGL_EXT_device_drm_render_node #0 /dev/dri/card1
```

Comment:
not sure if it's worth clarifying in the tutorial that the output here is from ogre2 probing the available EGL devices like you mentioned below. It could still be using GLX.  The `Starting EGL Subsystem` output is what confirms to me that ogre2 is using EGL.

Reply:
I've expanded this part of the tutorial.